### ISSUE TITLE
Race condition with async functions

### ISSUE THREAD

**Issue by AlexanderEpolite** [oven-sh/bun#6815]
### What version of Bun is running?

1.0.7+b0393fba6200d8573f3433fb0af258a0e33ac157

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

On this code:

```js
const fetch = require("node-fetch");

async function test() {
    
    return new Promise(async (resolve) => {
        fetch("http://127.0.0.1:8080", {
            headers: {
                "User-Agent": "real-ua/1.0.0"
            }
        }).then((async (r) => {
            const cookie = r.headers.raw()["set-cookie"][0].split("session=")[1].split(";")[0];
            
            await fetch("https://example.com", {
                headers: {
                    "Cookie": "session=" + cookie,
                    "User-Agent": "real-ua/1.0.0",
                }
            }).then(async (a) => {
                console.log("This will never execute on bun!")
            });
            
        }));
    });
    
}

test();
```

Running `node index.js` will result in "This will never execute on bun!", and running `bun index.js` will... not return anything!

What I have found that is VERY strange is that if the following is removed:
```
"Cookie": "session=" + cookie,
```
It will work on Bun, which leads me to believe this is some sort of race condition.

### What is the expected behavior?

Node and Bun should return the same result

### What do you see instead?

Bun does not return the same result and skips some parts of the code (where the console.log is).

### Additional information

test server:
```js

const http = require("http");

http.createServer((req, res) => {
    res.writeHead(200, "OK", {
        "set-cookie": "session=12345"
    });
    
    res.end("ok");
}).listen(8080);
```

Running on WSL 1.2.5.0 with an Intel 9600k.

Let me know if you need any other information.

**Comment by cirospaciari** [comment]
Did a little investigation:

```js
const fetch = require("node-fetch");

async function test() {
  return new Promise(async (resolve) => {
    fetch("http://127.0.0.1:8080", {
      headers: {
        "User-Agent": "real-ua/1.0.0",
      },
    }).then(async (r) => {
      try {
        const cookie = r.headers
          .raw()
          ["Set-Cookie"][0].split("session=")[1]
          .split(";")[0];

        await fetch("https://example.com", {
          headers: {
            Cookie: "session=" + cookie,
            "User-Agent": "real-ua/1.0.0",
          },
        }).then(async (a) => {
          console.log("This will never execute on bun!");
        });
      } catch (err) {
        console.error(err);
      }
    });
  });
}

await test();
```

Found out that `set-cookie` was returning from `raw()` with `Set-Cookie` key and the code actually throws an error, and nodejs actually returns `set-cookie` not `Set-Cookie`

**Comment by AlexanderEpolite** [comment]
So it's still a bug in Bun?

**Comment by cirospaciari** [comment]
> So it's still a bug in Bun?

Yeah soon will be fixed after https://github.com/oven-sh/bun/pull/7011

**Comment by cirospaciari** [comment]
resolved via #7011
